numpy/_core/include/numpy/npy_2_compat.h

```diff
@@ -74,11 +74,23 @@
     #define NPY_DEFAULT_INT NPY_INTP
     #define NPY_RAVEL_AXIS NPY_MIN_INT
     #define NPY_MAXARGS 64
+
+    static inline npy_uint64
+    PyDataType_FLAGS(const PyArray_Descr *dtype)
+    {
+        return dtype->flags;
+    }
 #elif NPY_ABI_VERSION < 0x02000000
     #define NPY_DEFAULT_INT NPY_LONG
     #define NPY_RAVEL_AXIS 32
     #define NPY_MAXARGS 32
 
+    static inline npy_uint64
+    PyDataType_FLAGS(const PyArray_Descr *dtype)
+    {
+        return dtype->flags;
```

Comment:
I think this needs an explicit cast to `npy_uint64`, because this is a signed char on numpy 1.x and might be negative?

Reply:
I added a cast to `unsigned char` instead.  I think so long we stick with unsigned we are fine anyway.  But the unsigned is a problem (opened a follow up issue and the release note snippet here)